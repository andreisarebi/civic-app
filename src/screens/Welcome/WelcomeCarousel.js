import React from 'react';
import PropTypes from 'prop-types';
import { Dimensions, StyleSheet, View } from 'react-native';
import Carousel, { Pagination } from 'react-native-snap-carousel';
import WelcomeCarouselItem from './WelcomeCarouselItem';
import { CandidatesImage, GetInvolvedImage, StayInformedImage } from './WelcomeImages';
import colors from '../../styles/colors';

class WelcomeCarousel extends React.Component {
  images = [
    {
      image: <CandidatesImage />,
      title: 'Learn where candidates stand on the issues',
    },
    {
      image: <StayInformedImage />,
      title: 'Stay informed about news that matters to you',
    },
    {
      image: <GetInvolvedImage />,
      title: 'Make an impact for causes you care about',
    },
  ];

  static propTypes = {
    isSmallScreen: PropTypes.bool,
  };

  constructor(props) {
    super(props);
    this.state = { currentIndex: 0 };
  }

  renderItem = ({ item, index }) => (
    <WelcomeCarouselItem key={index} image={item.image} isSmallScreen={this.props.isSmallScreen} title={item.title} />
  );

  render() {
    const screenWidth = Dimensions.get('window').width;
    return (
      <View style={styles.container}>
        <Carousel
          data={this.images}
          renderItem={this.renderItem}
          itemWidth={screenWidth}
          sliderWidth={screenWidth}
          useScrollView={true}
          onSnapToItem={index => this.setState({ currentIndex: index })}
          loop={true}
          autoplay={true}
          autoplayDelay={300}
          autoplayInterval={5000}
          lockScrollWhileSnapping={true}
        />
        {!this.props.isSmallScreen && (
        <Pagination
          activeDotIndex={this.state.currentIndex}
          dotsLength={this.images.length}
          dotColor={colors.white}
          inactiveDotColor={colors.lightBlue}
          inactiveDotScale={1}
          dotStyle={styles.dot}
        />
        )}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'flex-start',
    maxHeight: 500,
  },
  dot: {
    borderWidth: 1,
    borderColor: colors.white,
  },
});

export default WelcomeCarousel;
