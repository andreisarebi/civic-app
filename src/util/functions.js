import distanceInWordsStrict from 'date-fns/distance_in_words_strict';

export const distanceInWordsFromNow = (date) => {
  return distanceInWordsStrict(date, Date.now());
}
