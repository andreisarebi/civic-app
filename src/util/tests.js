import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { Provider } from 'react-redux';
import store from '../../App/configureStore';

export const createSnapshot = (Component, props) => {
  const renderer = new ShallowRenderer();
  renderer.render(<Component {...props} />);
  return renderer.getRenderOutput();
};

export const createConnectedSnapshot = (Component, props) => {
  const renderer = new ShallowRenderer();
  renderer.render(<Provider store={store}><Component {...props} /></Provider>);
  return renderer.getRenderOutput();
};
