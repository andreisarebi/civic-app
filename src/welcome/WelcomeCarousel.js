import React from 'react';
import { Dimensions, StyleSheet, View } from 'react-native';
import Carousel, { Pagination } from 'react-native-snap-carousel';
import WelcomeCarouselItem from './WelcomeCarouselItem';
import {
  CandidatesImage,
  GetInvolvedImage,
  StayInformedImage,
} from './WelcomeImages';
import colors from '../styles/colors';

class WelcomeCarousel extends React.Component {
  images = [
    {
      image: <CandidatesImage />,
      title: 'Learn About Candidates',
      subtitle: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit.',
    },
    {
      image: <StayInformedImage />,
      title: 'Stay Informed',
      subtitle: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit.',
    },
    {
      image: <GetInvolvedImage />,
      title: 'Get Involved',
      subtitle: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit.',
    },
  ];

  constructor(props) {
    super(props);
    this.state = { currentIndex: 0 };
  }

  renderItem = ({ item, index }) => (
    <WelcomeCarouselItem
      key={index}
      image={item.image}
      title={item.title}
      subtitle={item.subtitle}
    />
  );

  render() {
    const screenWidth = Dimensions.get('window').width;
    return (
      <View style={styles.container}>
        <Carousel
          data={this.images}
          renderItem={this.renderItem}
          itemWidth={screenWidth}
          sliderWidth={screenWidth}
          useScrollView={true}
          onSnapToItem={index => this.setState({ currentIndex: index })}
          loop={true}
          autoplay={true}
          autoplayDelay={300}
          autoplayInterval={5000}
          lockScrollWhileSnapping={true}
        />
        <Pagination
          activeDotIndex={this.state.currentIndex}
          dotsLength={this.images.length}
          dotColor={colors.white}
          inactiveDotColor={colors.lightBlue}
          inactiveDotScale={1}
          dotStyle={styles.dot}
          containerStyle={styles.paginationContainer}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  dot: {
    borderWidth: 1,
    borderColor: colors.white,
  },
  paginationContainer: {
    marginBottom: 24,
  },
});

export default WelcomeCarousel;
