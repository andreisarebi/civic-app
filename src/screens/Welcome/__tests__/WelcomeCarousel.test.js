import { createSnapshot } from '../../../util/tests';
import WelcomeCarousel from '../WelcomeCarousel';

describe('WelcomeCarousel', () => {
  it('renders correctly', () => {
    const snapshot = createSnapshot(WelcomeCarousel, { isSmallScreen: false });
    expect(snapshot).toMatchSnapshot();
  });
  it('does not render pagination in small screen', () => {
    const snapshot = createSnapshot(WelcomeCarousel, { isSmallScreen: true });
    expect(snapshot).toMatchSnapshot();
  });
});
