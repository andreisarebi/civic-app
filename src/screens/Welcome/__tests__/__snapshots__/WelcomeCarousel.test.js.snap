// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeCarousel does not render pagination in small screen 1`] = `
<Component
  style={
    Object {
      "flex": 1,
      "justifyContent": "flex-start",
      "maxHeight": 500,
    }
  }
>
  <Carousel
    activeAnimationOptions={null}
    activeAnimationType="timing"
    activeSlideAlignment="center"
    activeSlideOffset={20}
    apparitionDelay={0}
    autoplay={true}
    autoplayDelay={300}
    autoplayInterval={5000}
    callbackOffsetMargin={5}
    containerCustomStyle={Object {}}
    contentContainerCustomStyle={Object {}}
    data={
      Array [
        Object {
          "image": <CandidatesImage />,
          "title": "Learn where candidates stand on the issues",
        },
        Object {
          "image": <StayInformedImage />,
          "title": "Stay informed about news that matters to you",
        },
        Object {
          "image": <GetInvolvedImage />,
          "title": "Make an impact for causes you care about",
        },
      ]
    }
    enableMomentum={false}
    enableSnap={true}
    firstItem={0}
    hasParallaxImages={false}
    inactiveSlideOpacity={0.7}
    inactiveSlideScale={0.9}
    inactiveSlideShift={0}
    itemWidth={750}
    layout="default"
    lockScrollTimeoutDuration={1000}
    lockScrollWhileSnapping={true}
    loop={true}
    loopClonesPerSide={3}
    onSnapToItem={[Function]}
    renderItem={[Function]}
    scrollEnabled={true}
    shouldOptimizeUpdates={true}
    slideStyle={Object {}}
    sliderWidth={750}
    swipeThreshold={20}
    useScrollView={true}
    vertical={false}
  />
</Component>
`;

exports[`WelcomeCarousel renders correctly 1`] = `
<Component
  style={
    Object {
      "flex": 1,
      "justifyContent": "flex-start",
      "maxHeight": 500,
    }
  }
>
  <Carousel
    activeAnimationOptions={null}
    activeAnimationType="timing"
    activeSlideAlignment="center"
    activeSlideOffset={20}
    apparitionDelay={0}
    autoplay={true}
    autoplayDelay={300}
    autoplayInterval={5000}
    callbackOffsetMargin={5}
    containerCustomStyle={Object {}}
    contentContainerCustomStyle={Object {}}
    data={
      Array [
        Object {
          "image": <CandidatesImage />,
          "title": "Learn where candidates stand on the issues",
        },
        Object {
          "image": <StayInformedImage />,
          "title": "Stay informed about news that matters to you",
        },
        Object {
          "image": <GetInvolvedImage />,
          "title": "Make an impact for causes you care about",
        },
      ]
    }
    enableMomentum={false}
    enableSnap={true}
    firstItem={0}
    hasParallaxImages={false}
    inactiveSlideOpacity={0.7}
    inactiveSlideScale={0.9}
    inactiveSlideShift={0}
    itemWidth={750}
    layout="default"
    lockScrollTimeoutDuration={1000}
    lockScrollWhileSnapping={true}
    loop={true}
    loopClonesPerSide={3}
    onSnapToItem={[Function]}
    renderItem={[Function]}
    scrollEnabled={true}
    shouldOptimizeUpdates={true}
    slideStyle={Object {}}
    sliderWidth={750}
    swipeThreshold={20}
    useScrollView={true}
    vertical={false}
  />
  <Pagination
    activeDotIndex={0}
    dotColor="#ffffff"
    dotStyle={
      Object {
        "borderColor": "#ffffff",
        "borderWidth": 1,
      }
    }
    dotsLength={3}
    inactiveDotColor="#2d9cdb"
    inactiveDotOpacity={0.5}
    inactiveDotScale={1}
    tappableDots={false}
    vertical={false}
  />
</Component>
`;
